Build/Build.cs

```diff
@@ -56,12 +58,37 @@ class Build : RemotionBuild
         var packageJsonPath = ((IBaseBuild)this).Solution.Directory / "Remotion" / "Web" / "Dependencies.JavaScript" / "package.json";
         Assert.FileExists(packageJsonPath);
 
-        var outputFolder = ((IBaseBuild)this).OutputFolder / "Npm" / "remotion.dependencies" / "package.json";
-        var packageJsonContent = packageJsonPath.ReadAllText()
-                .Replace("$version$", ((IBuildMetadata)this).BuildMetadataPerConfiguration.First().Value.Version);
-        outputFolder.WriteAllText(packageJsonContent);
+        var outputPackageJson = ((IBaseBuild)this).OutputFolder / "Npm" / "remotion.dependencies" / "package.json";
+        AddVersionToPackageJson(packageJsonPath, outputPackageJson, ((IBuildMetadata)this).BuildMetadataPerConfiguration.First().Value.Version);
       });
 
+  public override ISbomGeneratorBuilder ConfigureSbomGenerationInfoBuilder (Solution solution)
+  {
+      var version = ((IBuildMetadata)this).GetBaseVersion();
+
+      var semanticVersion = SemanticVersion.Parse(version);
+      var shortenedVersion = $"{semanticVersion.Major}.{semanticVersion.Minor}.{semanticVersion.Patch}";
+
+      // We disregard test projects when generating the sbom already, so these are not required here.
+      var blacklistedProjects = new[]
+                                {
+                                    "Web.Dependencies.Javascript",
+                                    "*.Analyzers*"
+                                };
+
+      var packageJsonPath = solution.Directory / "Remotion" / "Web" / "Dependencies.JavaScript" / "package.json";
+      var packageJsonWithVersion = TemporaryDirectory / "sbom" / "package.json";
+
+      AddVersionToPackageJson(packageJsonPath, packageJsonWithVersion, shortenedVersion);
+
+      // We do not require github auth because we do not do enough requests for licenses and package infos
```

Comment:
Why do we not need to do this here but in the base implementation? I think I saw this somewhere with an acutal user/key. Once I know where and why we needed it there, I can better work with this comment :)

Reply:
Github requires auth when we send off too many requests to its API. The tool calls the github API in some cases. Since we don't really have that problem as our dependency count is low, we do not need to supply the auth info.

Reply:
okay, so the whole github-auth story is just to support this usecase but we never actually ran into it with re-motion?

Since we do have the auth information in our build (don't we?), shouldn't we still use github auth and just forget about the whole thing? Otherwise, we risk a future failing build if limits change, many builds run in a short time, etc.

Reply:
just saw the second comment about budget: if the properties are already available, we should add it (15 minutes or less). if we don't have them, yes, we can add later. As I said, I think we already did them in an earlier iteration, as far as I remember.